### PR TITLE
Bitgreen Carbon-Based Tokenomics

### PR DESCRIPTION
**Pull request by XeZZoR**
### Project Abstract

**A composable pallet for producing network tokens from the retirement of carbon credits.**

Our proposed model generates blockchain network tokens as a direct product of purchasing and retiring on-chain tokenized
carbon credits. With carbon as the objective measuring stick, we start by acknowledging that avoided GHG emission and
removed atmospheric CO2e are intrinsically valuable targets as a societal goal that has common good to perpetuate and
encourage through financial incentivization.

Unlike token pre-mines which are artificially scarce, high quality carbon sinks such as rainforests and mangroves, and
other natural resources including endangered species and thriving food stocks have natural scarcity. Nature has its own
balance that has become perilously dislodged since the industrial revolution. Most of these resources are becoming
increasingly more scarce every year as industry and consumption continue unabated.

Bitgreen will tokenize high quality carbon credits and offer them for sale and retirement on the Bitgreen parachain. A
user, upon purchasing and retiring a tokenized credit, receives in return an amount of Bitgreen Network Tokens (BNT)
determined by a dynamic tokenomics algorithm. BNT can be used to access the Bitgreen network, pay for certain
applications, vote on governance referendums, participate in staking, and pay transaction fees.

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying?

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $50,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] I have read and understood the [FAQs](https://github.com/w3f/Grants-Program/blob/master/docs/faq.md), [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md) and [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledge the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).
- [ ] I prefer the discussion of this application to be in a private Element/Matrix channel. My username is: ``@_______``

### How Did You Hear About our grants program?

- [ ] Social Media
- [ ] Hackathon
- [ ] Personal Recommendation
- [x] Substrate Builders Program
- [ ] Investor/VC
- [ ] Online Search
- [ ] Other: _______

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Just a reminder that everyone who contributed to the application needs to sign our terms and conditions via the CLA. Also could you add the default deliverables 0a. to 0d. also to the second milestone? Furthermore, something to be aware of is that we usually don’t fund security audits for projects that have their own tokens/aren’t common good projects. So could you explain the following a little bit more detailed: “Every BNT token is initially born from an original good deed.”? Maybe we can make an exception if the token isn’t used to fund the project itself.

**Comment by laboon** [comment]
How are you solving the oracle problem here (i.e. how are you verifying the offsetting/sequestration of carbon)? I apologize if this is explained in the accompanying images, but I am having trouble reading them. It appears that trusted third parties are used for this verification; if this is the case, then what exactly will the pallet be doing that can't be done by using Statemine/Statemint assets (or the assets pallet in general) and trusting the auditors ?

On a side note, I'm also curious if you have read Kim Stanley Robinson's "Ministry for the Future" which has the development of a "carbon coin" as a plot point. If not, you may be interested in doing so. =)

**Review by semuelle — CHANGES_REQUESTED** [review]
Some notes from me as well:

- Can you please host the images elsewhere? For example, if you paste them into the Github editor, they will automatically be available from githubusercontent.com
- Please have a look at which activities we fund, and which we don't in our [FAQ](https://github.com/w3f/Grants-Program/blob/master/docs/faq.md#what-activitiespositions-do-you-fund). Hosting and marketing and not part of them, so I'd suggest removing deliverable 2.3 and rephrasing 2.4.
- There is still a fair amount of template text in your application, please remove or replace it.
- A lot of the deliverable specifications are too vague. We ask for *verifiable* specs, so that there is no disagreement during evaluation. "Develop legal relationship" is not well defined.
- Your first milestone is 20,000 USD per person-month, the second 10,000. Can you explain the difference and why the price is that high?
- Since you mention Bitgreen tokens being paid out to the developers, can you share more about your tokenomics plans? Are these all CC-backed tokens?

**Comment by laboon** [comment]
Pinging @jonasW3F and @burdges

**Comment by XeZZoR** [comment]
> 

The credits that are in question here are already existing carbon credits that should be fully verified by the most common standards. Therefore there is not a direct Oracle issue here in the sence that we have to track sequestation of credits. 

We are mainly using the assets pallet to implement the tokenization of credits. However there are hooks needed based on the retirement (burning) of one asset to create another. The way the credits are minted needs to be authorized by Bitgreen staff currently due to limitations with technology on the standards itself. The VCU Pallet we already implemented with tokenization and retirement rules will need additional features that allow for the minting of BBB tokens on retirement.

However part of the roadmap that comes after this is to build a solution for continous monitioring applications. These mainly work on sattelite data that has verifiable inputs and open models. This is not part of the grant proposal though.

And yes we read that book at least a few of us did ;)

**Comment by laboon** [comment]
_The credits that are in question here are already existing carbon credits that should be fully verified by the most common standards. Therefore there is not a direct Oracle issue here in the sence that we have to track sequestation of credits.

We are mainly using the assets pallet to implement the tokenization of credits. However there are hooks needed based on the retirement (burning) of one asset to create another. The way the credits are minted needs to be authorized by Bitgreen staff currently due to limitations with technology on the standards itself. The VCU Pallet we already implemented with tokenization and retirement rules will need additional features that allow for the minting of BBB tokens on retirement._

Sorry, I still don't understand that first part and why it means that there's not a direct oracle issue. What are these most common standards? Are they stored on another blockchain somewhere? How are they verified?

**Comment by XeZZoR** [comment]
> How are you solving the oracle problem here (i.e. how are you verifying the offsetting/sequestration of carbon)? I apologize if this is explained in the accompanying images, but I am having trouble reading them. It appears that trusted third parties are used for this verification; if this is the case, then what exactly will the pallet be doing that can't be done by using Statemine/Statemint assets (or the assets pallet in general) and trusting the auditors ?

In the voluntary market there are 4 Major non profit "Standards" which are certifying bodies for issued carbon credits. Verra and Goldstandard are two of these with upwars of 90% of total market share on issued credits. They have centralized registries in which these credits can be tracked. The problem here is that no normal user will ever get an account into these in fact even we have trouble getting one.
The tracked carbon credits that we are going to tokenize in the first place need to be inside an escrow account that is not controlled by the seller (it will be controlled by us or a partner of ours). Currently there are manual steps required here for retirement and the tokenization will only be done after a human checks that the credits to tokenize were transferred to the escrow account.

As for the sequestration itself these Standards require strict methodologies to be followed before they certify any credits and we rely on these to be correct. In case of errors these standards have so called buffers that can be used to reimburse falsly accredited (and retired) carbon credits. These will be directly exchanged by the Standards. This hole complex is a little difficult to explain in short writing.

**Comment by XeZZoR** [comment]
> * Your first milestone is 20,000 USD per person-month, the second 10,000. Can you explain the difference and why the price is that high?

I will realign this. The first milestone is basically a research paper that leads into a Whitepaper in Milestone 2. There is a Professor named Will Cong and a Team of PHD students that are doing the research and whitepaper while Bitgreen is doing the implementation. The total cost for the research Whitepaper is 100.000$. The cost of implementation by the Bitgreen team are carried by ourselves.

I think the best is to realign these Milestones to put all the research into M1 and all development into M2 whereas M2 then basically has no cost.

**Comment by adamcarv** [comment]
> On a side note, I'm also curious if you have read Kim Stanley Robinson's "Ministry for the Future" which has the development of a "carbon coin" as a plot point. If not, you may be interested in doing so. =)

Bullseye. A portion of the genesis for Bitgreen was inspired by Ministry for the Future. We have talked to Delton Chen several times (quoted in the book as "The Chen Paper"). This research grant & tokenomics proposal is an attempt to actualize part of KSM's literary vision --  a currency whose scarcity model and intrinsic value derive from conserved natural resources. You nailed it =)

File: applications/Bitgreen_Carbon_Tokenomics.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+- **Costs:** 40000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0  |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and extensive module documentation. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Article | We will publish medium articles explaining how the Tokenomics system works **
+| 1a. | Relationships | Develop legal relationship with Will Cong based on published, peer-reviewed academic research on crypto-economics and consensus protocols. Co-design project parameters and negotiate consulting & licensing arrangement |  
+| 1b. | Documentation | Provide a research team with written project scope, performance indicators, timeline and final deliverables. Reach mutual agreement on scope |
```
[the remark this answers is not in the document]
Comment (XeZZoR) [reply applications/Bitgreen_Carbon_Tokenomics.md:236]: Redundant language will remove it.

**Review by semuelle — CHANGES_REQUESTED** [review]
Hey @XeZZoR & @adamcarv. Thanks for the many replies and updates. I'm generally interested in funding the project, and I understand your reasoning for shifting the costs to the first milestone. However, we are at least as interested in the implementation as in the preceding research, so I'd be more comfortable with the split you had previously. Not that I think you are not interested in completing the project, but we've had enough unfinished grants to err on the cautious side. Also, at this point M1 would be 100k for 2 person-months, something we have never paid before.

Lastly, one of your contributors hasn't [signed the CLA](https://github.com/w3f/Grants-Program/pull/972#issuecomment-1142085550) yet.

**Comment by adamcarv** [comment]
> Hey @XeZZoR & @adamcarv. Thanks for the many replies and updates. I'm generally interested in funding the project, and I understand your reasoning for shifting the costs to the first milestone. However, we are at least as interested in the implementation as in the preceding research, so I'd be more comfortable with the split you had previously. Not that I think you are not interested in completing the project, but we've had enough unfinished grants to err on the cautious side. Also, at this point M1 would be 100k for 2 person-months, something we have never paid before.
> 
> Lastly, one of your contributors hasn't [signed the CLA](https://github.com/w3f/Grants-Program/pull/972#issuecomment-1142085550) yet.

"Dennis" and XeZZor are the same person for signing :) 

Per the milestones, how about if we roll up all of the disbursements into one milestone based on implementation? That would completely de-risk the grant for W3F and make this project outcomes-based. That's acceptable to us as a proper carrot incentive.

**Comment by semuelle** [comment]
> how about if we roll up all of the disbursements into one milestone based on implementation?

It's an option, though we usually try to lower the risk (not the monetary one, but the risk of misunderstandings and disputes) by asking for small milestones, so that feedback and conversations happen along the way. 

I think the separation of milestones and payments is odd in your case because M1 is mainly about work between you and the research team. The grants program does pay for research and software development, but deliverables should be focused on what you are actually delivering to the ecosystem, not what is happening between two teams (e.g. "Proposal approval" and "Relationships"). Perhaps you can restructure the roadmap. 

I would also like to see a lot more technical details about the model and pallet. For example: 
- The `assets` pallet already provides the basic tokenization functionalities, but there must be either human intervention or some backend that tracks carbon credits on some off-chain platform(s)/accounts and keeps them in sync. How do you envision this? 
- What "other tokenized real world assets" are you planning to incorporate?
- What are these "attack vectors" you mention and what "security measures" are you thinking of needing/implementing?

**Comment by takahser** [comment]
@XeZZoR I'm quoting your previous comment:

> In the voluntary market there are 4 Major non profit "Standards" which are certifying bodies for issued carbon credits. Verra and Goldstandard are two of these with upwars of 90% of total market share on issued credits.

My questions here would be:

1. how your solution differs from [Toucan](https://toucan.earth/) that is using _Verra_ as their offsets verifier as well and previously had [issues with having lots of low-quality carbon credits](https://www.bloomberg.com/news/articles/2022-04-07/the-biggest-crypto-effort-to-end-useless-carbon-offsets-is-backfiring#xj4y7vzkg) on their platform
2. are you planning to put any guards in place to prevent low-quality carbon credits being minted on your platform

File: applications/Bitgreen_Carbon_Tokenomics.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+carbon credits. With carbon as the objective measuring stick, we start by acknowledging that avoided GHG emission and
+removed atmospheric CO2e are intrinsically valuable targets as a societal goal that has common good to perpetuate and
+encourage through financial incentivization.
+
+Unlike token pre-mines which are artificially scarce, high quality carbon sinks such as rainforests and mangroves, and
+other natural resources including endangered species and thriving food stocks have natural scarcity. Nature has its own
+balance that has become perilously dislodged since the industrial revolution. Most of these resources are becoming
+increasingly more scarce every year as industry and consumption continue unabated.
+
+Bitgreen will tokenize high quality carbon credits and offer them for sale and retirement on the Bitgreen parachain. A
```
[the remark this answers is not in the document]
Comment (semuelle) [reply applications/Bitgreen_Carbon_Tokenomics.md:95]: > we are not necessarily the owner at times of tokenization

> our plans are to take possesion of the credits in exchange for the tokens

These two sentences seem at odds to me. Can you elaborate?

**Comment by semuelle** [comment]
I only now realize that there are two different tokens, BNT and BBB. So BBB are tokenized CCs, and a "dynamic" amount of BNT is offered for burning BBBs?

**Comment by XeZZoR** [comment]
@semuelle when we wrote this we didn't have a Ticker so we took BNT as generic Bitgreen Network Token. By now we are having a ticker BBB and I left this here as is to avoid confusion which didn't work out cause we now introduced it ;) BBB and BNT are the same I will ammend the proposal to only mention BBB.

**Comment by XeZZoR** [comment]
> > how about if we roll up all of the disbursements into one milestone based on implementation?
> 
> It's an option, though we usually try to lower the risk (not the monetary one, but the risk of misunderstandings and disputes) by asking for small milestones, so that feedback and conversations happen along the way.
> 
> I think the separation of milestones and payments is odd in your case because M1 is mainly about work between you and the research team. The grants program does pay for research and software development, but deliverables should be focused on what you are actually delivering to the ecosystem, not what is happening between two teams (e.g. "Proposal approval" and "Relationships"). Perhaps you can restructure the roadmap.
> 
> I would also like to see a lot more technical details about the model and pallet. For example:
> 
> * The `assets` pallet already provides the basic tokenization functionalities, but there must be either human intervention or some backend that tracks carbon credits on some off-chain platform(s)/accounts and keeps them in sync. How do you envision this?
> * What "other tokenized real world assets" are you planning to incorporate?
> * What are these "attack vectors" you mention and what "security measures" are you thinking of needing/implementing?

The intention of this proposal is to offset our cost on this research matter. The Quote that we got from the research team is 100.000$ + BBB Tokens. Basically the grand here will be fully used by that Team. 

The struggle I have with adding more detail to the technical milestones here is that I frankly do not know the details yet. The research is kind of a requirement to have a clear vision on how to implement a well thought out model. The only detail that we do know is that we already developed a pallete that works with a marketplace to provide the offsets. This pallete certainly needs changes to incorporate this minting of BBB tokens based of the offset amounts. How that is supposed to work though and the full detail is subject to the model that we receive out of this research.

The provision this grand holds to the rest of the ecosystem is that the hole research will be public and can be used by anyone in the ecosystem. Our palletes are already public and can be used. The model is hopefully transferrable to other similar use-cases and the module developed based on it should be adoptable as well.

As to what other assets we plan to incorporate:
* Impact Investments
* Other Ecosystem Service Credits like Plastics, Biodiversity and RECs

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update here and all the quick replies. We discussed your grant application today internally. In general, it’s maybe best if we jump on a call to discuss it. Feel free to reach out to grants@web3.foundation to schedule something.  
One issue which we currently have is that we usually don’t fund anything related to the tokenecomics of a project or the initial whitepaper (as long as it’s not a pure common good project).

**Comment by burdges** [comment]
I'd meant to comment on this sooner, apologizes but I'm kinda allergic to this term "tokenomics"

A priori I'd expect a carbon credit system consists of project operator records, project funder tokens, and carbon credit token themselves.  At various points a project operators invokes an investigation fee payment, which acts like an oracle and if successful causes causes project funder tokens to emit carbon credit tokens.

We'd likely implement this logic on statemint anyways because it serves many use cases, but afaik it's just an airdrop really.  I'm unsure if we can prevent project funder tokens from being unregistered securities or if doing so matters or if you merely need an effective system for excluding Americans.

Investigations could even be funded by payments to an investigating service, perhaps on-chain but maybe off-chain fiat payments.  After investigating then the investigating service then triggers the on-chain emission of carbon credit tokens.  Investigations could be simple if they merely require that open source software inspect satellite photos.  Ideally you'd want highly respected naturalists to run the analysis software, but then make the assessment data available in github for other's comments, maybe they place git commits into the oracle calls or whatever, and then credits appear after some dispute period elapses.  

I believe the real question here should not be tokenomics so much as governance.  All of non-standard finance is soaked in fraud, but both the blockchain space and the carbon credit space especially, and some of it goes sideways badly ala [Land Life](https://twitter.com/web3isgreat/status/1549896775376445440), so who judges disputes?  We'd want carbon credits on polkadot to be respectable, ideally more so than whatever governments are promoting.

I suppose tokenomics appears if you want to limit who can raise disputes, like to silencing on-line trolls, but imho we should simply resolve this with credentialism plus scientific arguments.  In this, credentalism means in ecology or similar, never economics, because economists work around climate tends to be intellectual fraud, like Nordhaus' Nobel prize.
- https://www.youtube.com/watch?v=pGI0R1w_Xws
- https://www.researchgate.net/publication/344034609_The_appallingly_bad_neoclassical_economics_of_climate_change

There is imho work here for which grants funding makes sense, but it's tricky for me to call this work "tokenomics" per se.

**Comment by XeZZoR** [comment]
@cruikshankss thanks for your detailed reply and sorry for not answering sooner we are in preperation of a lot of things and I missed my notifications again ;) we will schedule something with the web3 Team. I think there is a lot of different smaller projects mentioned in this that led to confusion your points are well taken and valid concerns though

@burdges what you mention here is kind of a second project that we thought about for a longer time now and pitched to a few originators and brookers. The idea in general to create an on-chain tracked verification trail is something we intend to execute on in the not so distant future but currently miss the dev ressources for. This proposal is really about our governance token that we want to get into a utility token form. Happy to bounce of the carbon validator dao idea though in a separate conversation maybe we find a team to help us execute it ;)

**Comment by burdges** [comment]
Are there technical problems with investigations/assessment that need solving?  Or did you guys envision outsourcing those?

**Comment by XeZZoR** [comment]
> Afaik there is no need for verification to be tracked on chain per se, likely git repos suffice, but whatever track the assessments & disputes need proper backup/mirroring. And polkadot is _not_ a backup service.
> 
> Is there a need for a governance token? How would it be helpful? There is need for governance, but afaik the "respectability" goals exist entirely off chain, so likely governance should similarly exist off-chain.
> 
> Are there technical problems with investigations/assessment that need solving? Or do you guys envision outsourcing those?

The amount of people that are in the space that can really do an assesment of a project (Verifiers or VVBs as they are called) is very very low. Most of these are also specialized on a certain type of project i.E. forestry. There is a multitude of different methodologies and even though we have an inhouse expert on auditing the project documentations after they are verified the verification process itself is a bottleneck. The technical part of this piece would not be that hard to solve but getting the participants into this structure and finding enough is. 

This is a little vague as said I don't think this is really part of the proposal but I would be happy to give you a rundown if you are intested on another platform to not derail this too much.

**Comment by burdges** [comment]
We can chat sometime sure, maybe you can explain what governance token means here.

Are you guys on element/matrix?  We typically do long-term text chats there.  I've no signal or telegram device until early august.

**Comment by XeZZoR** [comment]
@dennis_bitgreen:matrix.org hit me up ;)

**Comment by adamcarv** [comment]
Thank you @burdges @cruikshankss @semuelle @laboon and others for the thoughtful comments, questions and feedback on BBB. I'd like to respond with an equally gracious reply, and will do so this weekend. 

In the meantime we would welcome a call next week to hear your thoughts. The conversation can cover the topic(s) of this grant application or whatever else may come up, as a few comments here have somewhat veered beyond the simple proposition we pose in our proposal. That question is:

- Is it a viable research project to investigate a tokenomics regime based on real-world assets with natural scarcity, as an alternative to traditional tokenomics based on synthetic, cryptocurrency assets buoyed by an arbitrarily-determined, programmable supply (e.g. 21M). 

That is the heart of this research project. Less so, carbon credits.

**Comment by alxs** [comment]
@XeZZoR as you can see, many of us are excited about the potential for climate action on Polkadot.

To add here to David's [comment](https://github.com/w3f/Grants-Program/pull/972#pullrequestreview-1037555029), I think there's a fundamental misunderstanding regarding the scope of the grants program and what we can and can't fund. Our focus lies on technical research and software development and we only fund deliverables that could be useful to others in the ecosystem. As such, we've never funded research into tokenomics for a specific token, and I don't think we can, no matter the project's goal. In case you need funding for it, that'd be better suited for treasury.

As such, I suggest you throughly rework your proposal. Ideally, you wouldn't mention the token at all. Our goal is not to invest in the project, but to fund deliverables that others can reuse. Looking at the individual deliverables, I think none of those in the first milestone fall within the scope of the grants program and those in the second one need to be described in significantly more detail. You can think of the deliverables as the requirements of a contract: we need to be able to assess whether your delivery meets the specification based on them.

I'm also happy to join the call with your team. Feel free to start rewriting the proposal, unless you prefer to wait until you have clearer technical requirements to apply. For now I'll put the application on hold.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Just to double check, should we start scheduling a call or will you reach out to  grants@web3.foundation? Apart from this, I just want to share this regarding Verra: 
https://www.ledgerinsights.com/carbon-credit-standards-body-verra-suspends-blockchain-crypto-tokenization/

**Comment by adamcarv** [comment]
> 

Verra's hands were tied and had to take this step due to the license some other blockchain companies were taking with old carbon credits. We are on leadership committees at IETA and Gold Standard to assist with the industry's transition to blockchain.

**Comment by adamcarv** [comment]
Thank you all for participating in the process. We've decided to withdraw the proposal

**Comment by Noc2** [comment]
Thanks for the update. I’m closing the application. Let us know if you want to work on anything else or need anything specific for your project. We could always create an RFP and try to find teams to implement it.

**Pull request closed without merging**